scalding-core/src/main/scala/com/twitter/scalding/reducer_estimation/Common.scala

```diff
@@ -181,6 +187,13 @@ object ReducerEstimatorStepStrategy extends FlowStepStrategy[JobConf] {
       // if still None, make it '-1' to make it simpler to log
       val numReducers = combinedEstimator.estimateReducers(info)
 
+      numReducers.foreach { n =>
+        val configuredMax = conf.getInt(EstimatorConfig.maxEstimatedReducersKey, EstimatorConfig.defaultMaxEstimatedReducers)
+        if (n > configuredMax) {
+          throw new IllegalArgumentException(s"Reducer estimator estimated $n reducers, which is more than the configured maximum of $configuredMax")
```

Comment:
Yes, let's not throw an exception to the user.

I don't think all kinds of cluster sizes can converge to one suitable default value for max number of reducers. As such I believe, we should have max number of reducers <= max number of mappers, presuming mappers are not exploding data.  For hadoop2 clusters, the config param mapreduce.jobtracker.split.metainfo.maxsize controls the number of mappers that can be launched. 
Not sure if it's possible in this code, but could it be possible to arrive at a suitable max value based on cluster config setting? For example, 1MB of mapreduce.jobtracker.split.metainfo.maxsize translates to roughly 10K map tasks. So we should set the number of reducers <= number of mappers (presuming mappers are not exploding input data). 
cc @lohitvijayarenu cc @gerashegalov